lib/rules/id-length.js

```diff
@@ -117,6 +117,8 @@ module.exports = {
                 return properties && !parent.computed && parent.key.name === node.name;
             },
             ImportDefaultSpecifier: true,
+            ImportSpecifier: true,
```

Comment:
I think `ImportSpecifier` should not be reported by the rule but the name assigned to it should.

```js
/*eslint id-length: ["error"]*/

import { x } from 'module';  // rule will not report 'x'

import { x as z } from 'module'; // rule will report report 'z'

```

Reply:
Let's keep the scope of this change limited to the accepted issue https://github.com/eslint/eslint/issues/18847. That is, only check `ImportNamespaceSpecifier`.

Reply:
Make sense to me! for this we can create an another issue, let me re-write the suggestions.